src/libasr/ASR.asdl

```diff
@@ -363,7 +363,7 @@ ttype
     | CPtr()
     | TypeParameter(identifier param, dimension* dims)
     | FunctionType(ttype* arg_types, ttype? return_var_type,
-        abi abi, deftype deftype, string? bindc_name, bool elemental,
+        abi abi, deftype deftype, string? bindc_name, string? c_header, bool elemental,
```

Comment:
`c_header` I think should go into `Function`, not be part of the type (neither logical nor physical: https://github.com/lfortran/lfortran/issues/1601).